src/jmespath/Lexer.js

```diff
@@ -299,7 +299,10 @@ export default class Lexer {
     }
     this._current += 1;
     const literal = stream.slice(start + 1, this._current - 1);
-    return literal.replaceAll("\\'", "'");
+    if (this._current > maxLength) {
```

Comment:
Seems like earlier unterminated string literals were supported. i.e "'Something" was valid expression. I assume that was unintentional ?

Reply:
Yes, unintentional.  We should be strict